src/main/java/homework5/FruitsStore.java

```diff
@@ -0,0 +1,56 @@
+package homework5;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class FruitsStore {
+    private ArrayList<Fruits> fruits = new ArrayList<>();
+    FruitsStore store;
+
+    public void add (Fruits f) {
+        this.fruits.add(f);
+    }
+
+    public List<Fruits> getFreshFruits(LocalDate date) {
+        return this.fruits.stream().filter(fruit ->
+                (fruit.getFreshnesTerm() >= this.lifeTimeCalc(date, fruit.getDeliveryDate())))
+                .collect(Collectors.toList());
+    }
+
+    public List<Fruits> getFruitsByType(FruitsEnum fEnum) {
+        return this.fruits.stream().filter(fruit ->
+                fruit.getFEnum() == fEnum).collect(Collectors.toList());
+    }
+
+    public List<Fruits> getFruitsByFreshnesTerm (int freshnessTerm) {
+        return fruits.stream().filter(fruit ->
+                fruit.getFreshnesTerm() > freshnessTerm).collect(Collectors.toList());
+    }
+
+    public List<Fruits> getFreshFruitsByType (FruitsEnum fEnum, LocalDate date) {
+        store = new FruitsStore();
+        store.fruits = new ArrayList<>(this.getFreshFruits(date));
+        return store.getFruitsByType(fEnum);
+    }
+
+    public void doSale (int realiztionTerm, LocalDate date, float percent, FruitsEnum... fEnums) {
+        for (FruitsEnum e : fEnums) {
+            this.fruits = new ArrayList<>(this.fruits.stream().map(fruit ->
+                    ((fruit.getFreshnesTerm() - this.lifeTimeCalc(date, fruit.getDeliveryDate())
+                            < realiztionTerm && (fruit.getFEnum() == e))
+                            ? new Fruits(fruit.getFreshnesTerm(), fruit.getDeliveryDate(), (int) (fruit.getPrice() * (percent/100)), fruit.getFEnum()) : fruit))
+                    .collect(Collectors.toList()));
```

Comment:
you change input List. Not good. Return new List
for every FruitsEnum e you create new list of results - it is mistake in logic

Reply:
there is no input list

Reply:
check out a change

Reply:
Your method change input list. fruits = new ArrayList<>(.... 
What will be happen when you use this method twice?
Read my comment 4 hour before: Return new List (now you have void)

Reply:
Thought it was a task to change the value, but not to return other List

Reply:
For you business newer say - create me new ArrayList. It your logic, in previuos case you can have many problems - one of methods changes input data for others